fastify.js

```diff
@@ -192,16 +196,20 @@ function build (options) {
   return fastify
 
   function routeHandler (req, res, params, context) {
-    res._context = context
     req.id = genReqId(req)
     req.log = res.log = log.child({ reqId: req.id, level: context.logLevel })
     req.originalUrl = req.url
 
     req.log.info({ req }, 'incoming request')
 
-    res._startTime = now()
-    res.on('finish', onResFinished)
-    res.on('error', onResFinished)
+    if (isAbstract === false) {
+      res._startTime = now()
```

Comment:
I think you can safely move `res._startTime = now()` inside the `if` below, since it is used only inside the `onResponseCallback` function.

Reply:
again, I would not touch this. The chances to have a change of shape is very high.

Reply:
Your suggestion is to remove all the `if` checks inside this function?

Reply:
no. Adding a property within an if is a clear strategy to cause deopts.

Reply:
done